tests/modules/programs/sbt/credentials.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  credentials = [
+    {
+      realm = "Sonatype Nexus Repository Manager";
+      host = "example.com";
+      user = "user";
+      password = "password";
+    }
+    {
+      realm = "Sonatype Nexus Repository Manager X";
+      host = "v2.example.com";
+      user = "user1";
+      password = "password1";
+    }
+  ];
+  expectedCredentialsSbt = pkgs.writeText "credentials.sbt" ''
+    credentials += Credentials("Sonatype Nexus Repository Manager", "example.com", "user", "password")
+    credentials += Credentials("Sonatype Nexus Repository Manager X", "v2.example.com", "user1", "password1")
+  '';
+  credentialsSbtPath = ".sbt/1.0/credentials.sbt";
+in {
+  config = {
+    programs.sbt = {
+      enable = true;
+      credentials = credentials;
+      package = pkgs.runCommandLocal "sbt" { pname = "sbt"; } ''
+        mkdir -p $out/bin
+        touch $out/bin/sbt
+        chmod +x $out/bin/sbt;
+      '';
```

Comment:
```suggestion
      package = pkgs.writeScriptBin "dummy-sbt" "";
```

Reply:
Oh, nice. I copied this from vscode tests, I suppose I can update it there as well in #1762.